### PR TITLE
 Feature/api create user show endpoint

### PR DESCRIPTION
**Pull request by mtakac2**

### CODE REVIEW

**Comment by mtakac2** [comment]
@manwithtwowatches this PR is changing some of the json attributes (see updated specs https://app.apiary.io/opapi/editor)

**Comment by ulferts** [comment]
@mtakac @manwithtwowatches  we probably shouldn't merge this then until the front end can handle this, right?

File: lib/api/v3/work_packages/work_package_representer.rb [hunk 1]
```diff
@@ -51,6 +51,27 @@ def initialize(options = {}, *expand)
           { href: "#{root_url}/api/v3/work_packages/#{represented.work_package.id}", title: "#{represented.subject}" }
         end
 
+        link :author do
+            {
+                href: "#{root_url}/api/v3/users/#{represented.work_package.author.id}",
+                title: "#{represented.work_package.author.name} - #{represented.work_package.author.login}"
+            } unless represented.work_package.author.nil?
+        end
```
Comment (myabc) [inline lib/api/v3/work_packages/work_package_representer.rb:59]: @mtakac why not move the `unless` one line down?

Comment (mtakac2) [inline lib/api/v3/work_packages/work_package_representer.rb:59]: @myabc okay

**Comment by mtakac2** [comment]
@ulferts no we should not. I've checked with @manwithtwowatches and it should be easy fix on the client side.

**Comment by mtakac2** [comment]
@tilt so these needs to be merged to get access to the latest API version (according to the API specs)

**Comment by tilt** [comment]
@mtakac Thanks! I'll take a look at it but will not base my latest overview tab stuff on it before Monday

File: public/templates/tabs/overview.html [hunk 2]
```diff
@@ -26,11 +26,28 @@
     <li><label>Status</label>{{ workPackage.props.status }}</li>
     <li><label>Priortiy</label>{{ workPackage.props.priority }}</li>
     <li><label>Date</label>{{ workPackage.props.startDate }} - {{ workPackage.props.dueDate }}</li>
-    <li><label>Responsible</label><img class="avatar" ng-src="{{ workPackage.props.responsibleAvatar }}" /><span  class="user"><a href="#">{{ workPackage.props.responsibleName }}</a></span>
```
Comment (tilt) [inline public/templates/tabs/overview.html:29]: these changes have to be rebased on current dev

File: public/templates/tabs/overview.html [hunk 3]
```diff
@@ -54,23 +71,39 @@
...
-        <span class="user"><a href name="{{currentNote}}" ng-bind="activity.props.userName"></a></span>
-        <span class="date">commented on <span ng-bind="activity.props.createdAt | date:'short'"/></span>
-        <span class="comment">
-          <ul class="work-package-details-activities-messages">
-            <li ng-repeat="message in activity.props.messages track by $index">
-              <span class="message" ng-bind="message"/>
-            </li>
-          </ul>
-        </span>
+    <li>
```
Comment (tilt) [inline public/templates/tabs/overview.html:74]: why has the correct implementation been replaced by the mock again?

File: public/templates/tabs/overview.html [hunk 4]
```diff
@@ -54,23 +71,39 @@
...
   </ul>
 </div>
+
+<div class="comments-form">
+  <h3>Add your comments here</h3>
+  <form>
+    <textarea placeholder="Add comments here" rows="4"></textarea>
+  </form>
+  <button class="button">Add comment</button>
+</div>
```
Comment (tilt) [inline public/templates/tabs/overview.html:109]: should be removed again for now

Comment (tilt) [inline public/templates/tabs/overview.html:109]: the file was moved to work_packages/tabs anyway, so this should be deleted before merging back

File: public/templates/tabs/overview.html (hunk 2 above)
Comment (myabc) [inline public/templates/tabs/overview.html:29]: I tried to rebase on dev and was unable to:

```
c7ab071 Added some properties to the API user representer
db5b99e Added API show user endpoint
021a673 Added links and embedded resources to the work package API resource
31bd77d Got rid of some inline WP properties
5168470 Got rid of authorization for show user API endpoint
828adb5 Added API show user endpoint
fad50e2 Added links and embedded resources to the work package API resource
f8619af Just moved some stuff around
1a3b2a9 Added API show user endpoint
d491a6c Added links and embedded resources to the work package API resource
cac8acc Got rid of authorization for show user API endpoint
bc753c8 Added links and embedded resources to the work package API resource
b164ac6 Just moved some stuff around
aa7e84a Corrected some errors in API work package representer
2aa42e6 Show assignee & responsible in the WP detail overview
dbc8681 Added some properties to the API user representer
c234e23 Added API show user endpoint
4c5562e Added links and embedded resources to the work package API resource
d0e92a7 Got rid of some inline WP properties
0845090 Got rid of authorization for show user API endpoint
26a75af Added links and embedded resources to the work package API resource
ab00b91 Just moved some stuff around
a65d015 Added some properties to the API user representer
a382548 Added API show user endpoint
e33f0bd Added links and embedded resources to the work package API resource
a15a47e Got rid of some inline WP properties
b7a51da Got rid of authorization for show user API endpoint
950c7fe Added some properties to the API user representer
cf30f52 Added API show user endpoint
6ec5e42 Added links and embedded resources to the work package API resource
1850976 Added some properties to the API user representer
29f1acf Got rid of some inline WP properties
e3a98c5 Got rid of authorization for show user API endpoint
b355721 Just moved some stuff around
4e5b974 Corrected some errors in API work package representer
5b0a821 Show assignee & responsible in the WP detail overview
```

Comment (tilt) [inline public/templates/tabs/overview.html:29]: I moved this file and it returned at some point, would that help doing the rebase? Or let's do a merge instead

**Comment by tilt** [comment]
From the API perspective looking at the code this looks good to me. But the overview tab isn't correctly set up.

**Pull request closed without merging**